avocado/core/test.py

```diff
@@ -1093,13 +1094,32 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
         :param locations: list of URLs from where the asset can be
                           fetched (optional)
         :param expire: time for the asset to expire
-        :raise EnvironmentError: When it fails to fetch the asset
-        :returns: asset file local path
+        :param find_only: whether it should only look for the asset in the
+                          cache and do not download/move it to the cache.
+        :param cancel_on_missing: whether the test should be canceled if the
+                                  asset was not found in the cache. Takes
+                                  effect only when parameter `find_only` is
+                                  `True`.
+        :raises: :class:`exceptions.TestCancel when `find_only` is `True` and
+                 the asset was not found in the cache.
+        :raises EnvironmentError: When it fails to fetch the asset
+        :returns: asset file local path or `None` when asset was not found in
+                  the cache.
         """
         if expire is not None:
             expire = data_structures.time_to_seconds(str(expire))
-        return asset.Asset(name, asset_hash, algorithm, locations,
-                           self.cache_dirs, expire).fetch()
+
+        asset_obj = asset.Asset(name, asset_hash, algorithm, locations,
+                                self.cache_dirs, expire)
+
+        if find_only:
+            asset_path = asset_obj.find_asset_file()
```

Comment:
Before, the `find_asset_file` was meant to be used independently. Since @clebergnu suggested it could be a feature of the the `fetch_asset` here, it seems the knowledge of what method to call should be in the `Asset` class itself. I'd just parametrize the `Asset.fetch()` call with the `find_only` option and restrict the scope of the change here.
Anyway, not a big deal. I'm happy when @clebergnu is happy.

Reply:
> Before, the `find_asset_file` was meant to be used independently. Since @clebergnu suggested it could be a feature of the the `fetch_asset` here, it seems the knowledge of what method to call should be in the `Asset` class itself. I'd just parametrize the `Asset.fetch()` call with the `find_only` option and restrict the scope of the change here.

I tried to still keep the Asset class a standalone and free from _test_ terminology module to be used out of a test, keeping each behavior in its own method.

> Anyway, not a big deal. I'm happy when @clebergnu is happy.

That's the objective :D

Reply:
Fair enough, thanks!